models/reverse_index.py

```diff
@@ -0,0 +1,76 @@
+import pickle
+
+
+class ReverseIndex(object):
+    """
+    This class contains our model for our reverse index
+    + attributes:
+        - reverse_index: contains the reverse index with the following structure:
+            {term: [frequence_doc, [document_ids...]]}
+    + methods:
+        - add_document: update self.reverse_index with a given document
+        - save: save the current self.reverse_index in the given filename
+        - load_from_file: load the reverse index contains in the given file
+        - create_index: create reverse_index from given document_collection
+    """
+
+    def __init__(self):
+        """
+        We init the reverse_index attribute to an empty dict.
+        """
+        self.reverse_index = {}
+
+    def add_document(self, document):
+        """
+        Update self.reverse_index given a document
+        + params:
+            - document: should contain a term_bag property built like this:
+                {term, frequence}
+        + return:
+            None
+        + summary:
+            foreach term contains in keys of term_bag document:
+                we update the frequence_doc of current term
+                we add the document_id
+        """
```

Comment:
Maybe we should verify that the term_bag property exists here ?

Reply:
Currently if the term_bag property does not exist, it will raise an AttributeError. What behaviour do we want instead?
- Raise a custom error?
- Not block the program execution? In this case should we log the caught error?
- Any other idea?

Reply:
Ok I did not realized that, I think AttributeError is okay